### PR TITLE
Model classes

### PR DESCRIPTION
**Pull request by ezraermy**
In this pull request, I:
- Created model classes
- Inserted data based on created models
- Created custom methods
- Checked the data based on the custom models

#### Other requirements
- No linter errors
- Follow gitflow 
- Documented professional way

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  def update_user_post_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] [OPTIONAL]
It's a good practice to use one of the Rails Active Callbacks `after_save` to call the `update_post_counter` method every time a new post has been saved to the database, and when `after_save` is called mark the `update_post_counter` method as _private_ accordingly, which should be similar to:

```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Doing so will enable the triggering of update logic at the right instance.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (rloterh) [inline app/models/comment.rb]: - [ ] [OPTIONAL]
Similarly, it's a good practice to use one of the Rails Active Callbacks `after_save` to call the `update_comment_counter` method every time a new comment has been saved to the database, and when `after_save` is called mark the `update_comment_counter` method as _private_ accordingly, which should be similar to:

```ruby
 # app/models/comment.rb
  after_save :update_comment_counter

  private

  def update_comment_counter
    author.increment!(:comments_counter)
  end
```

Doing so will enable the triggering of update logic at the right instance.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post
+
+  def update_post_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (rloterh) [inline app/models/like.rb]: - [ ] [OPTIONAL]
Again, it's a good practice to use one of the Rails Active Callbacks `after_save` to call the `update_like_counter` method every time a new like has been saved to the database, and when `after_save` is called mark the `update_like_counter` method as _private_ accordingly, which should be similar to:

```ruby
 # app/models/like.rb
  after_save :update_like_counter

  private

  def update_like_counter
    author.increment!(:likes_counter)
  end
```

Doing so will enable the triggering of update logic at the right instance.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Ezra,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job establishing data models associations!
🎲 Good ReadMe!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 3 above)
Comment (ezraermy) [reply app/models/like.rb]: Hi @rloterh, thanks a lot for the feedback and tips indeed!

**Pull request merged**